### PR TITLE
[Feature-8138][Task] add at@ msg in the dingtalk task plugin

### LINKED ISSUE #8138: [Feature][Task] add at@ msg in the dingtalk task plugin
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

add at@ msg in the dingtalk task plugin

When a custom bot sends a message, you can specify the "@person list" by your mobile phone number. When the people in the "@people list" receive the message, there will be a @ message reminder. Do not disturb conversations still notify reminders, and "someone @ you" appears on the fold.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of the pull request

this pr close #8138

image

iShot2022-01-20 18 04 43

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/DingTalkAlertChannelFactory.java [hunk 1]
```diff
@@ -64,6 +64,28 @@ public String name() {
...
+                .addValidate(Validate.newBuilder()
+                        .setRequired(false)
+                        .build())
+                .build();
+        InputParam atUserIdsParam = InputParam
+                .newBuilder(DingTalkParamsConstants.NAME_DING_TALK_AT_USERIDS, DingTalkParamsConstants.DING_TALK_AT_USERIDS)
+                .addValidate(Validate.newBuilder()
+                        .setRequired(false)
+                        .build())
+                .build();
```
Comment (zhongjiajie) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/DingTalkAlertChannelFactory.java:78]: Does mention users by id work for dingtalk now?

**Comment by zhongjiajie** [comment]
BTW, could you please add document to dingtalk alert ?

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/DingTalkAlertChannelFactory.java (hunk 1 above)
Comment (zhuangchong) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/DingTalkAlertChannelFactory.java:78]: The document provided by the official website has this field
```
https://open.dingtalk.com/document/robots/custom-robot-access
```
I tested the mobile phone number, but I didn't find out what the userId of DingTalk is, so I didn't test the userId

**Comment by zhuangchong** [comment]
> BTW, could you please add document to dingtalk alert ?

I'll update the DingTalk alert document right away.

**Review by zhongjiajie — DISMISSED** [review]
LGTM, waiting CI

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

**Comment by zhongjiajie** [comment]
Please add doc in the same level of https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/alert/enterprise-wechat.html if you have time, thanks

**Pull request merged**